File: .sdk/tm/py/utility/make_error.py
# ChineseMedicineClinics SDK utility: make_error

from __future__ import annotations
from core.operation import ChineseMedicineClinicsOperation
from core.result import ChineseMedicineClinicsResult
from core.control import ChineseMedicineClinicsControl
from core.error import ChineseMedicineClinicsError


def make_error_util(ctx, err):
    if ctx is None:
        from core.context import ChineseMedicineClinicsContext
        ctx = ChineseMedicineClinicsContext({}, None)

    op = ctx.op
    if op is None:
        op = ChineseMedicineClinicsOperation({})
    opname = op.name
    if opname == "" or opname == "_":
        opname = "unknown operation"

    result = ctx.result
    if result is None:
        result = ChineseMedicineClinicsResult({})
    result.ok = False

    if err is None:
        err = result.err
    if err is None:
        err = ctx.make_error("unknown", "unknown error")

    errmsg = ""
    if isinstance(err, ChineseMedicineClinicsError):
        errmsg = err.msg
    elif hasattr(err, "msg") and err.msg is not None:
        errmsg = err.msg
    elif isinstance(err, str):
        errmsg = err
    else:
        errmsg = str(err)

    msg = "ChineseMedicineClinicsSDK: " + opname + ": " + errmsg
    msg = ctx.utility.clean(ctx, msg)

    result.err = None

    spec = ctx.spec

    if ctx.ctrl.explain is not None:
        ctx.ctrl.explain["err"] = {"message": msg}

    sdk_err = ChineseMedicineClinicsError("", msg, ctx)
    sdk_err.result = ctx.utility.clean(ctx, result)
    sdk_err.spec = ctx.utility.clean(ctx, spec)

    if isinstance(err, ChineseMedicineClinicsError):
        sdk_err.code = err.code

    ctx.ctrl.err = sdk_err

    if ctx.ctrl.throw_err is False:
        return result.resdata

    raise sdk_err

File: php/utility/MakeError.php
<?php
declare(strict_types=1);

// ChineseMedicineClinics SDK utility: make_error

require_once __DIR__ . '/../core/Operation.php';
require_once __DIR__ . '/../core/Result.php';
require_once __DIR__ . '/../core/Error.php';

class ChineseMedicineClinicsMakeError
{
    public static function call(?ChineseMedicineClinicsContext $ctx, mixed $err): mixed
    {
        if ($ctx === null) {
            require_once __DIR__ . '/../core/Context.php';
            $ctx = new ChineseMedicineClinicsContext([], null);
        }
        $op = $ctx->op ?? new ChineseMedicineClinicsOperation([]);
        $opname = $op->name;
        if ($opname === '' || $opname === '_') {
            $opname = 'unknown operation';
        }

        $result = $ctx->result ?? new ChineseMedicineClinicsResult([]);
        $result->ok = false;

        if ($err === null) {
            $err = $result->err;
        }
        if ($err === null) {
            $err = $ctx->make_error('unknown', 'unknown error');
        }

        $errmsg = ($err instanceof ChineseMedicineClinicsError) ? $err->msg : (string)$err;
        $msg = "ChineseMedicineClinicsSDK: {$opname}: {$errmsg}";
        $msg = ($ctx->utility->clean)($ctx, $msg);

        $result->err = null;
        $spec = $ctx->spec;

        if ($ctx->ctrl->explain) {
            $ctx->ctrl->explain['err'] = ['message' => $msg];
        }

        $sdk_err = new ChineseMedicineClinicsError('', $msg, $ctx);
        $sdk_err->result = ($ctx->utility->clean)($ctx, $result);
        $sdk_err->spec = ($ctx->utility->clean)($ctx, $spec);
        if ($err instanceof ChineseMedicineClinicsError) {
            $sdk_err->sdk_code = $err->sdk_code;
        }

        $ctx->ctrl->err = $sdk_err;

        if ($ctx->ctrl->throw_err === false) {
            return $result->resdata;
        }
        throw $sdk_err;
    }
}
